systemFlake.nix

```diff
@@ -66,13 +69,19 @@ let
       extraArgs = sharedExtraArgs // lhs.extraArgs // rhs.extraArgs;
     };
 
-  foldHosts = foldl'
-    (lhs: rhs:
-      let output = rhs.name; in
-      # manually merge whats inside the output to prevent // override
-      lhs // { ${output} = lhs.${output} or { } // rhs.value; }
-    )
-    { };
+  mergeAny = lhs: rhs:
+    lhs // mapAttrs
+      (name: value:
+        if isAttrs value then lhs.${name} or { } // value
```

Comment:
We could make this recursive and I think we pretty much just get a local copy of `recursiveUpdate`. I'm not yet sure if this is safe however and if there was a reason for nixpkgs lib not implementing recursiveUpdate like this. But its also not necessary, recursion is only needed for otherArguments merging, but the `mergeArgs` below might be the safer solution.

Reply:
Can you confirm this accurately replaces `mergeHosts`?

Reply:
Yup I've tested it with the examples you setup - really useful thanks for setting that up - and the merging is handled well.